### PR TITLE
check cli output for kpt fn eval

### PR DESCRIPTION
**Pull request by natasha41575**
Fix https://github.com/GoogleContainerTools/kpt/issues/1666

### CODE REVIEW

File: pkg/test/runner/runner.go [hunk 1]
```diff
@@ -311,7 +311,10 @@ func (r *Runner) compareResult(exitErr error, stdout string, stderr string, tmpP
 func (r *Runner) compareOutput(stdout string, stderr string) error {
 	expectedStderr := r.testCase.Config.StdErr
 	if expectedStderr == "" && stderr != "" {
-		return fmt.Errorf("unexpected stderr %s", stderr)
+		// special case for fn-eval/invalid-fn-config-file
```
Comment (Shell32-Natsu) [inline pkg/test/runner/runner.go:314]: You can change the content of invalid config file in `fn-eval/invalid-fn-config-file` to what you prefer.

File: pkg/test/runner/runner.go [hunk 2]
```diff
@@ -142,17 +142,17 @@ func (r *Runner) runFnEval() error {
 		}
 	}
 	for i := 0; i < r.testCase.Config.RunCount(); i++ {
-		output, _, fnErr := runCommand("", r.kptBin, kptArgs)
+		stdout, stderr, fnErr := runCommand("", r.kptBin, kptArgs)
 		if fnErr != nil {
-			r.t.Logf("kpt error output: %s", output)
+			r.t.Logf("kpt error output: %s", stdout)
```
Comment (Shell32-Natsu) [inline pkg/test/runner/runner.go:147]: We may need to print stderr here as well.

File: pkg/test/runner/runner.go (hunk 1 above)
Comment (natasha41575) [reply pkg/test/runner/runner.go:314]: The problem here is that we can't get the stderr from the config file when it is invalid, so there needs to be a special case for it. 

Why do we have invalid-fn-config-file? Is it to test the harness itself? Is it necessary to keep it?

Comment (Shell32-Natsu) [reply pkg/test/runner/runner.go:314]: What is invalid is the function config file not test config file. So I am not sure why can't we get the stderr here.

Comment (natasha41575) [reply pkg/test/runner/runner.go:314]: ah, my mistake. I misread the files and thought that was the test config file. I'll fix that, thanks!

File: pkg/test/runner/runner.go (hunk 2 above)
Comment (natasha41575) [reply pkg/test/runner/runner.go]: fixed

File: e2e/testdata/fn-eval/invalid-fn-config-file/pkg/.expected/config.yaml [hunk 3]
```diff
@@ -16,3 +16,4 @@ testType: eval
 exitCode: 1
 image: gcr.io/kpt-fn/set-namespace:v0.1
 fnConfig: ../../config.yaml
+stdErr: "wrong Node Kind for  expected: MappingNode was ScalarNode: value: {I am not a valid config file}"
```
Comment (frankfarzan) [inline e2e/testdata/fn-eval/invalid-fn-config-file/pkg/.expected/config.yaml:19]: this is a pretty bad error message, it's not really comprehensible or actionable: go/kpt-errors-style-guide
It's leaking low-level YAML constructs to the end-user.

Can you file an issue for fixing `set-namespace` (it's probably comes from underlying kyaml library though).

Comment (frankfarzan) [inline e2e/testdata/fn-eval/invalid-fn-config-file/pkg/.expected/config.yaml:19]: Formatting: `for  expected` (two spaces).

File: e2e/testdata/fn-eval/missing-fn-config/.expected/config.yaml [hunk 4]
```diff
@@ -15,3 +15,4 @@
 testType: eval
 exitCode: 1
 image: gcr.io/kpt-fn/set-namespace:v0.1
+stdErr: "failed to configure function: input namespace cannot be emptyerror: exit status 1"
```
Comment (frankfarzan) [inline e2e/testdata/fn-eval/missing-fn-config/.expected/config.yaml:18]: line break issue.

As part of adding tests, can you also audit the error messages for content and style? That's the spirit of this issue.

go/kpt-errors-style-guide

File: e2e/testdata/fn-eval/invalid-fn-config-file/pkg/.expected/config.yaml (hunk 3 above)
Comment (frankfarzan) [reply e2e/testdata/fn-eval/invalid-fn-config-file/pkg/.expected/config.yaml:19]: Having low-level YAML errors is not necessary bad if it's qualified and wordsmitehd as such:

`YAML has an issue: expected a map, but got a scalar value `{I am ...}`

File: e2e/testdata/fn-eval/missing-fn-config/.expected/config.yaml (hunk 4 above)
Comment (natasha41575) [reply e2e/testdata/fn-eval/missing-fn-config/.expected/config.yaml:18]: Will do

File: e2e/testdata/fn-eval/invalid-fn-config-file/pkg/.expected/config.yaml (hunk 3 above)
Comment (natasha41575) [reply e2e/testdata/fn-eval/invalid-fn-config-file/pkg/.expected/config.yaml:19]: I'll file an issue for this and other bad error messages. Thank you for the feedback

File: e2e/testdata/fn-eval/missing-fn-config/.expected/config.yaml (hunk 4 above)
Comment (natasha41575) [reply e2e/testdata/fn-eval/missing-fn-config/.expected/config.yaml:18]: Could you clarify what you mean by line break issue? I don't see anything about line breaks in the errors style guide

Comment (frankfarzan) [reply e2e/testdata/fn-eval/missing-fn-config/.expected/config.yaml:18]: what's `emptyerror`?

Comment (frankfarzan) [reply e2e/testdata/fn-eval/missing-fn-config/.expected/config.yaml:18]: Taking a step back, if we put our end-users' hats on: does this sentence make sense?

Comment (natasha41575) [reply e2e/testdata/fn-eval/missing-fn-config/.expected/config.yaml:18]: Ah, understood. Thank you.

Comment (frankfarzan) [reply e2e/testdata/fn-eval/missing-fn-config/.expected/config.yaml:18]: Not resolved?

Comment (natasha41575) [reply e2e/testdata/fn-eval/missing-fn-config/.expected/config.yaml:18]: Resolved, thanks. Filed the issue here https://github.com/GoogleContainerTools/kpt/issues/1793

**Review by frankfarzan — APPROVED** [review]

**Pull request merged**